src/main/java/com/google/api/codegen/transformer/OutputTransformer.java

```diff
@@ -398,7 +398,47 @@ public void run() {
             valueSet.getId(),
             config.input());
       } else if (token == '{') {
-        throw new UnsupportedOperationException("map indexing not supported yet");
+        Preconditions.checkArgument(
```

Comment:
Could you add a comment/TODO referencing #2600 , that the subscript syntax may change?